pkg/catalog/diagram.go

```diff
@@ -80,9 +89,32 @@ func (p *Generator) CreateIntegrationDiagram(m *sysl.Module, title string, EPA b
 	return p.CreateIntegrationDiagramPlantuml(m, title, EPA)
 }
 
-//TODO @ashwinsajiv: fill out this function to generate mermaid integration diagrams
 func (p *Generator) CreateIntegrationDiagramMermaid(m *sysl.Module, title string, EPA bool) string {
-	return "" //p.CreateFile("plantumlString", mermaidjs, "title", "integration.Output+p.Ext")
+	var result string
+	var err error
+	var apps []string
+	for app := range m.Apps {
+		apps = append(apps, app)
+	}
+	mod := p.RootModule
+	if EPA {
+		result, err = endpointanalysisdiagram.GenerateMultipleAppEndpointAnalysisDiagram(mod, apps)
+		if err != nil {
+			p.Log.Error(err)
+			return ""
+		}
+	} else {
+		result, err = integration.GenerateMultipleAppIntegrationDiagram(mod, apps)
+		if len(apps) == 0 {
```

Comment:
Shouldn't this be checked before `if EPA`?

Reply:
Good catch, I'll move it to the front!